core/src/main/java/com/linecorp/armeria/common/websocket/WebSocketFrame.java

```diff
@@ -0,0 +1,369 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.websocket;
+
+import static com.linecorp.armeria.common.websocket.PingWebSocketFrame.emptyPing;
+import static com.linecorp.armeria.common.websocket.PongWebSocketFrame.emptyPong;
+import static java.util.Objects.requireNonNull;
+
+import java.nio.charset.StandardCharsets;
+
+import com.linecorp.armeria.common.BinaryData;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.unsafe.PooledObjects;
+
+import io.netty.buffer.ByteBuf;
+
+/**
+ * A {@link WebSocketFrame} to send data to the peer.
+ *
+ * @see <a href="https://datatracker.ietf.org/doc/html/rfc6455#section-5">Data framing</a>
+ */
+@UnstableApi
+public interface WebSocketFrame extends BinaryData {
+
+    /**
+     * Returns a new text {@link WebSocketFrame} with the text whose {@code finalFragment}
+     * is set to {@code true}.
+     *
+     * @see <a href="https://datatracker.ietf.org/doc/html/rfc6455#section-5.6">Data Frames</a>
+     */
+    static TextWebSocketFrame ofText(String text) {
```

Comment:
`TextWebSocketFrame` is package private and it doesn't add any public API. 
```suggestion
    static WebSocketFrame ofText(String text) {
```